### ISSUE TITLE
Invariant Violation: _registerComponent(...): Target container is not a DOM element.

### ISSUE THREAD

**Issue by miraten** [meteor/react-packages#12]
I created a new app with name 'my-react-app'.
And I cloned the 'react-packages' in the project directory with the name 'pakages'.
Then, I executed the command 'meteor add react'. I worked well.

Trying to run the app, I got the message:
'Invariant Violation: _registerComponent(...): Target container is not a DOM element.'

How can I resolve it?

My PC settings:
- Mac OS X Yosemite 10.10.3
- Meteor 1.1.0.2

**Comment by AdamBrodzinski** [comment]
Can you elaborate what your doing in between adding the package and the error? It sounds like the place you're trying to render to doesn't exist. If you can link to an example repo that's even more helpful.

Also are you using the `react-template-helper` package to render your component?

Here's a minimal example (with template helper) working as of 76d20f:

```html
// my-react-app.html

<head>
  <title>my-react-app</title>
</head>

<body>
  <h1>Welcome to Meteor!</h1>

  {{> hello}}
</body>

<template name="hello">
  <button>Click Me</button>
  <p>You've pressed the button {{counter}} times.</p>

  {{> React component=HelloWorld }}
</template>     
```

```jsx
// HelloWorld.jsx

HelloWorld = React.createClass({
  render() {
    return (
      <div>
        Hello World!
      </div>
    );
  },
});
```

**Comment by miraten** [comment]
What I did is as follows:

meteor create my-react-app
cd my-react-app
git clone https://github.com/meteor/react-packages
mv react-packages packages
meteor add react
rm my-react-app.js

And then, I edited 'my-react-app.html' and wrote a new file 'my-react-app.jsx' as your minimal example codes.

Running the app, the error in the browser console:
'Uncaught Error: No such template: React'

So I added the package 'react-template-helper'.
Running the app, the error in the brower console:

```
Exception from Tracker afterFlush function:
Error: In template hello, call to `{{> React ... }}` missing `component` argument.
    at null.<anonymous> (react-template-helper.js:11)
    at view.js:191
    at Function.Template._withTemplateInstanceFunc (template.js:437)
    at view.js:190
    at Object.Blaze._withCurrentView (view.js:523)
    at viewAutorun (view.js:189)
    at Tracker.Computation._compute (tracker.js:294)
    at new Tracker.Computation (tracker.js:210)
    at Object.Tracker.autorun (tracker.js:533)
    at Blaze.View.autorun (view.js:201)
```

the 'leaderboard' app also shows the same error.
However, 'react-todo' app works well.

**Comment by AdamBrodzinski** [comment]
It sounds like you're getting closer. The helper can't render because the component argument is undefined. I bet the file load order is different in our examples. Try renaming the jsx file to `a-my-react-app.jsx` to see if that fixes it. 

You can also pass the component in with a Blaze helper function that just returns the React component. I believe this helper would run after Meteor.isReady but i'm not 100% sure.

**Comment by stubailo** [comment]
I know that I get `Target container is not a DOM element.` when I try to call `React.render` when the page is not ready. Try putting your `render` call inside `Meteor.startup`.

**Comment by miraten** [comment]
Hi stubailo,

Thank you for your comment.

I got resolved the problem by putting it inside `Meteor.startup`.

And I found another problem and fixed it.

before:
```
      // Access the data from trackMeteorData() on this.data
      this.data.tasks.map(function (task) {
        return <li>task.content</li>
      })
```

after:
```
      // Access the data from trackMeteorData() on this.data
      this.data.tasks.map(function (task) {
        return <li key={task._id}>{task.content}</li>
      })
```

**Comment by stubailo** [comment]
Thanks for the feedback! Sorry for not actually running that code before writing it down.

**Comment by stubailo** [comment]
Fixed: ba7cb6747b08de4c9ecdb427879d08dcc3d6c209
